sdk/core/core-paging/review/core-paging.api.md

```diff
@@ -4,13 +4,24 @@
 
 ```ts
 
+// @public
+export function getPagedAsyncIterator<TOptions extends Record<string, any>, TResponse, TElement, TPage = TElement[]>(pagedResult: PagedResult<TOptions, TResponse, TPage>, path: string, options: TOptions): PagedAsyncIterableIterator<TElement, TPage>;
+
 // @public
 export interface PagedAsyncIterableIterator<T, PageT = T[], PageSettingsT = PageSettings> {
     [Symbol.asyncIterator](): PagedAsyncIterableIterator<T, PageT, PageSettingsT>;
     byPage: (settings?: PageSettingsT) => AsyncIterableIterator<PageT>;
     next(): Promise<IteratorResult<T, T>>;
 }
 
+// @public
+export interface PagedResult<TOptions extends Record<string, any>, TResponse, TPage> {
+    buildPage?: (response: TResponse) => TPage;
+    getNextLink?: (response: TResponse) => string | undefined;
+    maxPageSizeParam?: string;
```

Comment:
yeah that is a good point but I have to know that param name in case the customer wants to use the default `byPage` method which sets that parameter from the input settings.

Reply:
I ended up doing what you suggested and got around setting the `maxPageSize` issue by having it an explicit para in `getPage`.